clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -618,6 +618,10 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
                     // between prev_instr_ and memref.
                     shard->prev_xfer_marker_.marker.marker_type ==
                         TRACE_MARKER_TYPE_KERNEL_EVENT ||
+                    // Allow a CPU ID marker between a branch, and the branch
+                    // target.
+                    (shard->prev_entry_.marker.type == TRACE_TYPE_MARKER &&
+                     shard->prev_entry_.marker.marker_type == TRACE_MARKER_TYPE_CPU_ID) ||
```

Comment:
The current check is all about 'saw_timestamp_but_no_instr_`.  Relaxing on a cpuid seems to defeat the purpose of the original check as cpuid is pretty much always after a timestamp.  It seems that either this whole check needs to be re-thought and done in a different way, or -- and this seems the case to me -- delaying branches is supposed to never hit and it seems like a bug in raw2trace.

Reply:
It's indeed a corner case not being handled in raw2trace. I've updated the PR accordingly.